presto-main/src/main/java/com/facebook/presto/server/TaskInfoResource.java

```diff
@@ -63,10 +95,58 @@ public TaskInfo getTaskInfo(@PathParam("taskId") TaskId taskId)
                     }
                 }
             }
-            throw new NotFoundException("TaskInfo not found for task id " + taskId.toString());
+            throw new NotFoundException("TaskInfo not found for task id " + taskId);
         }
         catch (Exception e) {
             throw new NotFoundException(e);
         }
     }
+
+    @GET
+    @Path("{taskId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public void getTaskInfo(@PathParam("taskId") TaskId taskId,
+            @QueryParam("includeLocalQueryOnly") @DefaultValue("false") boolean includeLocalQueryOnly,
+            @HeaderParam(X_FORWARDED_PROTO) String xForwardedProto,
+            @Context UriInfo uriInfo,
+            @Context HttpServletRequest servletRequest,
+            @Suspended AsyncResponse asyncResponse)
+            throws WebApplicationException
+    {
+        try {
+            if (resourceManagerEnabled && !dispatchManager.isQueryPresent(taskId.getQueryId()) && !includeLocalQueryOnly) {
```

Comment:
nit : I would add a small method which captures the essence of this check for readability.
the resulting code should something like 
```
if (requestNeedsToBeProxied(..))
{
//proxy
} 
else {
//resume
}
```